solidity/contracts/hooks/axelar/AxelarHook.sol

```diff
@@ -0,0 +1,120 @@
+    using StandardHookMetadata for bytes;
+    using BridgeAggregationHookMetadata for bytes;
+    using Message for bytes;
+
+    IAxelarGasService public immutable AXELAR_GAS_SERVICE;
+    IAxelarGateway public immutable AXELAR_GATEWAY;
+    string public DESTINATION_CHAIN;
+    string public DESTINATION_CONTRACT;
+    bytes GMP_CALL_DATA;
+
+    constructor(
+        address _mailbox,
+        string memory destinationChain,
+        string memory destionationContract,
+        address axelarGateway,
+        address axelarGasReceiver,
+        bytes memory gmp_call_data
+    ) MailboxClient(_mailbox) {
+        DESTINATION_CHAIN = destinationChain;
+        DESTINATION_CONTRACT = destionationContract;
+        AXELAR_GATEWAY = IAxelarGateway(axelarGateway);
+        AXELAR_GAS_SERVICE = IAxelarGasService(axelarGasReceiver);
+        GMP_CALL_DATA = gmp_call_data;
+    }
+
+    /**
+     * @notice Returns an enum that represents the type of hook
+     */
+    function hookType() external pure returns (uint8) {
+        return uint8(IPostDispatchHook.Types.AXELAR);
+    }
+
+    /**
+     * @notice Returns whether the hook supports metadata
+     * @return true the hook supports metadata
+     */
+    function supportsMetadata(bytes calldata) external pure returns (bool) {
+        return true;
+    }
+
+    function postDispatch(
+        bytes calldata metadata,
+        bytes calldata message
+    ) external payable {
+        // ensure hook only dispatches messages that are dispatched by the mailbox
+        bytes32 id = message.id();
+        require(_isLatestDispatched(id), "message not dispatched by mailbox");
+
+        bytes memory axelarPayload = _formatPayload(message);
+        // Pay for gas used by Axelar with ETH
+        AXELAR_GAS_SERVICE.payNativeGasForContractCall{value: msg.value}(
+            address(this),
+            DESTINATION_CHAIN,
+            DESTINATION_CONTRACT,
+            axelarPayload,
+            metadata.refundAddress(address(0))
+        );
+
+        // bridging call
+        AXELAR_GATEWAY.callContract(
+            DESTINATION_CHAIN,
+            DESTINATION_CONTRACT,
+            axelarPayload
+        );
+    }
+
+    /**
+     * @notice Post action after a message is dispatched via the Mailbox
+     * @param metadata The metadata required for the hook. Metadata should contain custom metadata
+     *                 adhering to the BridgeAggregationHookMetadata structure.
+     */
+    function quoteDispatch(
+        bytes calldata metadata,
+        bytes calldata
+    ) external pure returns (uint256) {
+        bytes calldata bridgeMetadata = metadata.getCustomMetadata();
+
+        uint256 quote = bridgeMetadata.axelarGasPayment();
+        require(quote > 0, "No Axelar Payment Received");
+
+        return quote;
+    }
+
+    function _formatPayload(
+        bytes calldata message
+    ) internal view returns (bytes memory) {
+        return abi.encodePacked(GMP_CALL_DATA, message.id());
```

Comment:
Okay, this should be something like this:
```abi.encodePacked(VERSION, payload)```
where VERSION is 4 bytes (like 0x0001?)
and payload is 
```
{
  "submit_meta": {
     "id": "hyperlane_message_id"
  }
}
```

Reply:
modifying the contract to compute the payload internally instead for clarity on testing. Good chance we revert back to pre-computed GMP_CALL_DATA generated on construct for some gas savings

Reply:
just for the record, [this comment](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3050#discussion_r1428078595) is now obsolete, we agreed on using different version of payload encoding (version 1)